tests/test_cookiecutter_generation.py

```diff
@@ -60,33 +93,49 @@ def test_default_configuration(cookies, context):
     check_paths(paths)
 
 
-@pytest.fixture(params=["use_mailhog", "use_celery", "windows"])
-def feature_context(request, context):
-    context.update({request.param: "y"})
-    return context
-
-
-def test_enabled_features(cookies, feature_context):
-    result = cookies.bake(extra_context=feature_context)
-    assert result.exit_code == 0
-    assert result.exception is None
-    assert result.project.basename == feature_context["project_slug"]
-    assert result.project.isdir()
-
-    paths = build_files_list(str(result.project))
-    assert paths
-    check_paths(paths)
-
-
-def test_flake8_compliance(cookies):
+@pytest.mark.parametrize("windows", BINARY_CHOICES)
+@pytest.mark.parametrize("use_docker", BINARY_CHOICES)
+@pytest.mark.parametrize("use_celery", BINARY_CHOICES)
+@pytest.mark.parametrize("use_mailhog", BINARY_CHOICES)
+@pytest.mark.parametrize("use_sentry", BINARY_CHOICES)
+@pytest.mark.parametrize(
+    # These 2 cannot be used together, but test the other combinations
+    ["use_compressor", "use_whitenoise"],
+    [("y", "n"), ("n", "n"), ("n", "n")],
+)
```

Comment:
I'm not really satisfied with this duplication of parameters. If we want to avoid it, it looks like we need to combine [fixtures and parametrization](https://docs.pytest.org/en/latest/fixture.html#fixture-parametrize) or use the [`pytest_generate_tests` hook](https://docs.pytest.org/en/latest/parametrize.html#pytest-generate-tests).